### DISCUSSION TITLE
S3 access from WASM notebook?

### DISCUSSION THREAD

**Discussion by tmchartrand** [marimo-team/marimo#2437] [Q&A] [answered]
I read this in one of your blog posts on the WASM/pyodide notebooks and got very excited: "We are working on our own implementation of the Emscripten filesystem built on top of any S3-compatible bucket. This means you’ll be able to list, read, and write files from your S3 bucket just by interacting with ”os” filesystem all in Python."

Is this feature still in progress? any idea when it might be available?

**Comment by mscolnick** [comment, marked as answer]
@tmchartrand - on my phone so i may be a bit brief: 

We do have this now - it's not self-hostable, but it is free on the community cloud: https://marimo.io/sign-up

If you create a WASM project, you can add "integrations" which include connecting an S3 bucket. You can also upload your own files to each project and we provide a bucket for you

**Reply by tmchartrand** [reply]
excellent, I'd looked in the docs and the playground but not made an account yet! opening an issue about empty credentials for public buckets now...

**Comment by tmchartrand** [comment]
Also meant to ask @mscolnick, is there any chance this feature will be self-hostable in the future?

**Reply by mscolnick** [reply]
Maybe in the long-term, but not likely in the short-term just due to prioritization and likely additional support
